### PR TITLE
Leica m4 tomerge remoteproc

### PR DESCRIPTION
**Pull request by Qingwu-Li**

### CODE REVIEW

**Comment by Qingwu-Li** [comment]
Please have a review on this PR!

File: arch/arm64/configs/crocodile_defconfig [hunk 1]
```diff
@@ -1235,3 +1235,5 @@ CONFIG_MTD_CMDLINE_PARTS=y
 CONFIG_MTD_CHAR=y
 CONFIG_MTD_BLKDEVS=y
 CONFIG_JFFS2_FS=y
+CONFIG_IMX_REMOTEPROC=y
+
```
Comment (htho-lgs) [inline arch/arm64/configs/crocodile_defconfig:1239]: Please remove additional new line

File: arch/arm64/boot/dts/freescale/crocodile-rpmsg.dtsi [hunk 2]
```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2020 NXP
+ * Copyright (C) 2021 Leica Geosystems AG
+ */
+
+#include <dt-bindings/usb/pd.h>
+#include "crocodile-pinmap.h"
```
Comment (htho-lgs) [inline arch/arm64/boot/dts/freescale/crocodile-rpmsg.dtsi:8]: Why is it required?

File: arch/arm64/boot/dts/freescale/crocodile.dtsi [hunk 3]
```diff
@@ -300,7 +300,7 @@
 	status = "okay";
 
 	uhf@0 {
-		compatible = "spidev";
+		compatible = "lgs-crocodile,spi";
```
Comment (htho-lgs) [inline arch/arm64/boot/dts/freescale/crocodile.dtsi:303]: These changes are not related to remote proc.

**Review by htho-lgs — CHANGES_REQUESTED** [review]

**Review by htho-lgs — APPROVED** [review]

**Review by zandrey — CHANGES_REQUESTED** [review]
Please use `git cherry-pick -x` for external commits, this way we can track where they are coming from.

**Comment by Qingwu-Li** [comment]
@zandrey now all picked with -x, please review!

File: arch/arm64/boot/dts/freescale/crocodile-rpmsg.dtsi [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+		vdev0buffer: vdev0buffer@b8400000 {
+			compatible = "shared-dma-pool";
+			reg = <0 0xb8400000 0 0x100000>;
+			no-map;
+		};
+	};
+
+	imx8mm-cm4 {
+		compatible = "fsl,imx8mm-cm4";
+		clocks = <&clk IMX8MM_CLK_M4_DIV>;
```
Comment (emfend) [inline arch/arm64/boot/dts/freescale/crocodile-rpmsg.dtsi:42]: Since commit 811e4171 ('clk: imx: imx8mm: use imx8m_clk_hw_composite_core') IMX8MM_CLK_M4_DIV is only left for backwards compatibility.
So I guess it's better to use the new IMX8MM_CLK_M4_CORE instead.

**Comment by emfend** [comment]
Why are these commits required?
remoteproc: imx_rproc: Move common structure to header file
remoteproc: imx_rproc: Change to ioremap_wc for dram

**Review by htho-lgs — APPROVED** [review]

**Review by emfend — APPROVED** [review]
Looks good to me now.
